crates/turborepo-lib/src/daemon/connector.rs

```diff
@@ -0,0 +1,664 @@
+
+        #[cfg(not(target_os = "windows"))]
+        let make_service = move |_| {
+            // we clone the reference counter here and move it into the async closure
+            let path = path.clone();
+            async move { tokio::net::UnixStream::connect(path.as_path()).await }
+        };
+
+        #[cfg(target_os = "windows")]
+        let make_service = move |_| {
+            let path = path.clone();
+            async move { win(path) }
+        };
+
+        // note, this endpoint is just a dummy. the actual path is passed in
+        Endpoint::try_from("http://[::]:50051")
+            .expect("this is a valid uri")
+            .connect_with_connector(tower::service_fn(make_service))
+            .await
+            .map(TurbodClient::new)
+            .map_err(DaemonConnectorError::Socket)
+    }
+
+    /// Kills a currently active server but shutting it down and waiting for it
+    /// to exit.
+    async fn kill_live_server(
+        &self,
+        client: DaemonClient<()>,
+        pid: sysinfo::Pid,
+    ) -> Result<(), DaemonConnectorError> {
+        if client.stop().await.is_err() {
+            self.kill_dead_server(pid).await?;
+        }
+
+        match timeout(
+            Self::SHUTDOWN_TIMEOUT,
+            wait_for_file(&self.pid_file, WaitAction::Deleted),
+        )
+        .await?
+        {
+            Ok(_) => Ok(()),
+            Err(_) => self.kill_dead_server(pid).await,
+        }
+    }
+
+    /// Kills a server that is not responding.
+    async fn kill_dead_server(&self, pid: sysinfo::Pid) -> Result<(), DaemonConnectorError> {
+        let lock = self.pid_lock();
+
+        let system = sysinfo::System::new_with_specifics(
+            RefreshKind::new().with_processes(ProcessRefreshKind::new()),
+        );
+
+        let owner = lock
+            .get_owner()
+            .and_then(|p| system.process(sysinfo::Pid::from(p as usize)));
+
+        // if the pidfile is owned by the same pid as the one we found, kill it
+        match owner {
+            Some(owner) if pid == owner.pid() => {
+                debug!("killing dead server with pid: {}", pid);
+                owner.kill();
+
+                Ok(())
+            }
+            Some(owner) => {
+                debug!("pidfile is owned by another process, ignoring");
+                Err(DaemonConnectorError::WrongPidProcess(pid, owner.pid()))
+            }
+            // pidfile has no owner and has been cleaned up so we're ok
+            None => Ok(()),
+        }
+    }
+
+    async fn wait_for_socket(&self) -> Result<(), DaemonConnectorError> {
+        timeout(
+            Self::SOCKET_TIMEOUT,
+            wait_for_file(&self.sock_file, WaitAction::Exists),
+        )
+        .await?
+        .map_err(Into::into)
+    }
+
+    fn pid_lock(&self) -> pidlock::Pidlock {
+        pidlock::Pidlock::new(self.pid_file.clone().into_path_buf())
+    }
+}
+
+#[cfg(target_os = "windows")]
+fn win(
+    path: Arc<PathBuf>,
```

Comment:
Fixship: windows tests are failing, I think this is now an `AbsoluteNormalizedPath`